internal/initwd/module_install.go

```diff
@@ -254,7 +258,40 @@ func (i *ModuleInstaller) moduleInstallWalker(ctx context.Context, manifest mods
 					}
 
 					log.Printf("[TRACE] ModuleInstaller: Module installer: %s %s already installed in %s", key, record.Version, record.Dir)
-					return mod, record.Version, diags
+
+					var modDeprecation *configs.ModuleDeprecationInfo
+
+					var moduleVersion *response.ModuleVersion
+
+					// Checking for module deprecations in the case no new module versions need installation
+					if addr, isRegistryModule := req.SourceAddr.(addrs.ModuleSourceRegistry); isRegistryModule {
+						regClient := i.reg
+
+						regsrcAddr := regsrc.ModuleFromRegistryPackageAddr(addr.Package)
+						resp, err := regClient.ModuleVersions(ctx, regsrcAddr)
+						if err != nil {
+							log.Printf("[DEBUG] Deprecation for %s could not be checked: call to registry failed", addr.Package.Namespace)
+
+						} else {
+							found := false
+							for _, modProviderVersions := range resp.Modules {
+								for _, modVersions := range modProviderVersions.Versions {
+									vm, _ := version.NewVersion(modVersions.Version)
+									if vm.Equal(record.Version) {
+										moduleVersion = modVersions
+										found = true
+										break
+									}
+								}
+								if found {
+									break
+								}
+							}
+						}
+					}
+
+					modDeprecation = collectModuleDeprecationWarnings(moduleVersion, req.Name, record.Version)
```

Comment:
Think it should be fine since inside `collectModuleDeprecationWarnings` I have a nil check like so:

```go
func collectModuleDeprecationWarnings(moduleVersion *response.ModuleVersion, sourceName string, version *version.Version) *configs.ModuleVersionDeprecationInfo {
	var registryModDeprecation *configs.RegistryModuleVersionDeprecation

	if moduleVersion != nil && moduleVersion.Deprecation != nil { <---- HERE
		registryModDeprecation = &configs.RegistryModuleVersionDeprecation{
			Link:    moduleVersion.Deprecation.Link,
			Message: moduleVersion.Deprecation.Reason,
			Version: version.Original(),
		}
	}
	return &configs.ModuleVersionDeprecationInfo{
		SourceName:           sourceName,
		RegistryDeprecation:  registryModDeprecation,
		ExternalDependencies: []*configs.ModuleVersionDeprecationInfo{},
	}
}
```

Correct me if I'm wrong 😁

Reply:
For greater context I have it like this so there's always a parent `ModuleVersionDeprecationInfo` to attach any external dependencies to. So like the case of a local module that has external dependencies in the registry.

Reply:
Ah understood! This issue was indirectly called out elsewhere– fix is Included here https://github.com/hashicorp/terraform/pull/34943/commits/06454c33637d854352e84a857ace02a4b975fad5